projects/packages/my-jetpack/_inc/components/product-card/use-pricing-data.ts

```diff
@@ -0,0 +1,57 @@
+import { __ } from '@wordpress/i18n';
+import { PRODUCT_STATUSES } from '../../constants';
+import useProduct from '../../data/products/use-product';
+import { ProductCamelCase } from '../../data/types';
+
+const parsePricingData = ( pricingForUi: ProductCamelCase[ 'pricingForUi' ] ) => {
+	const { tiers } = pricingForUi;
+
+	if ( pricingForUi.tiers ) {
+		const { discountPrice, fullPrice, currencyCode } = tiers.upgraded;
+		const hasDiscount = discountPrice && discountPrice !== fullPrice;
+		return {
+			discountPrice: hasDiscount ? discountPrice / 12 : null,
+			fullPrice: fullPrice / 12,
+			currencyCode,
+		};
+	}
+
+	const { discountPricePerMonth, fullPricePerMonth, currencyCode, isIntroductoryOffer } =
+		pricingForUi;
+	return {
+		discountPrice: isIntroductoryOffer ? discountPricePerMonth : null,
+		fullPrice: fullPricePerMonth,
+		currencyCode,
+	};
+};
+
+const getPurchaseAction = ( detail: ProductCamelCase ) => {
+	if ( detail.status === PRODUCT_STATUSES.CAN_UPGRADE ) {
+		return __( 'Upgrade', 'jetpack-my-jetpack' );
+	}
+	if ( ! [ PRODUCT_STATUSES.ACTIVE ].includes( detail.status ) ) {
+		return __( 'Purchase', 'jetpack-my-jetpack' );
+	}
+
+	return null;
+};
+
+const getLearnMoreAction = ( detail: ProductCamelCase ) => {
+	if ( detail.status !== PRODUCT_STATUSES.ACTIVE && detail.tiers.includes( 'free' ) ) {
```

Comment:
Not all products that have a free offering have the "free" tier in the product detail. You might want to also check the `has_free_offering` and potentially even the `requires_plan` value that is currently only used on the backend. Just to make sure there is no free offering as Backup (as far as I can remember) should really be the only one without one 🤔

Reply:
Hmm, good point, however, here I wanted to reflect how our interstitials look like (as this is the label of button leading to interstitial). E.g. VideoPress, Stats, Social don't display "Start for free" option in the interstitial, they are also following "tiers" logic from what I understand.

But I'm not completely sure which way we should go here. 🤔

Reply:
> VideoPress, Stats, Social don't display "Start for free" option in the interstitial, they are also following "tiers" logic from what I understand.

This is true! However with the updated statuses PR, all of those plans (if they are in the owned section at least) will have an "Activate" button rather than a "Learn More" -> interstitial CTA reducing friction as much as possible. The goal was to allow users to activate and use our free offering if it exists as easily as possible, and some of the interstitials don't offer the free version on them as they should. That will be handled when we redo interstitials but for now I think it'd be best to allow them to start for free as easily as possible, especially if we are specifically recommending it to them

Reply:
For what it's worth, this isn't a problem for most of these product cards. The only ones I can see that are missing it are VideoPress, CRM, and Social 😄

Reply:
I think Boost as well. It's good point, I'll update the condition here to include all products that have free offering